py/desitarget/targets.py

```diff
@@ -336,64 +336,77 @@ def initial_priority_numobs(targets, survey='main'):
             import desitarget.sv1.sv1_targetmask as targmask
         if survey == 'sv2':
             import desitarget.sv2.sv2_targetmask as targmask
-        colnames = ["{}_{}_TARGET".format(survey.upper(),tc) for tc in ["DESI", "BGS", "MWS"]]
+        colnames = ["{}_{}_TARGET".format(survey.upper(), tc) for tc in ["DESI", "BGS", "MWS"]]
         masks = [targmask.desi_mask, targmask.bgs_mask, targmask.mws_mask]
     else:
         log.critical("survey must be either 'main', 'cmx' or 'sv', not {}!!!"
                      .format(survey))
 
-    #ADM set up the output arrays
-    outpriority = np.zeros(len(targets), dtype='int')-1
-    outnumobs = np.zeros(len(targets), dtype='int')-1
-
-    for colname, mask in zip(colnames,masks):
-        #ADM first determine which bits actually have priorities
+    # ADM much of the code is set up to expect NUMOBS==2 for "just" an LRG
+    # ADM but we need NUMOBS==2 for a 1-PASS LRG. The easiest way to deal
+    # ADM with this is just to unset the LRG bit if the LRG sub-bits are set.
```

Comment:
I think you can avoid this special case handling by setting `numobs['desi_mask']['LRG'] = 0` in targetmask.yaml so that the `LRG_1PASS` and `LRG_2PASS` bits will determine numobs instead of the `LRG` bit.

If that doesn't work, then we have a different logic error, since numobs should be the max of the numobs for any of the individual bits.

Reply:
I initially set the code up exactly as you're suggesting (with `NUMOBS` being set to 0 for the `LRG` bit). I decided I was concerned about the possibility that an LRG slipped through the cracks somehow (perhaps due to future code changes) without `LRG_1PASS` or `LRG_2PASS` being set. On reflection, though, I think I'll do it your way, but add an assert in `cuts.py` that tests for that case.

Reply:
Fixed in the latest push.